internal/resource/config/sp/sp_authentication_policy_contract_mapping_resource.go

```diff
@@ -0,0 +1,285 @@
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while creating the SpAuthenticationPolicyContractMappingResource", err, httpResp)
+		return
+	}
+	_, responseErr := spAuthenticationPolicyContractMappingResponse.MarshalJSON()
+	if responseErr != nil {
+		diags.AddError("There was an issue retrieving the response of the SpAuthenticationPolicyContractMappingResource: %s", responseErr.Error())
+	}
+
+	// Read the response into the state
+	var state spAuthenticationPolicyContractMappingResourceModel
+
+	diags = readSpAuthenticationPolicyContractMappingResourceResponse(ctx, spAuthenticationPolicyContractMappingResponse, &state)
+	resp.Diagnostics.Append(diags...)
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *spAuthenticationPolicyContractMappingResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state spAuthenticationPolicyContractMappingResourceModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	apiReadSpAuthenticationPolicyContractMappingResource, httpResp, err := r.apiClient.SpAuthenticationPolicyContractMappingsAPI.GetApcToSpAdapterMappingById(config.ProviderBasicAuthContext(ctx, r.providerConfig), state.Id.ValueString()).Execute()
+
+	if err != nil {
+		if httpResp != nil && httpResp.StatusCode == 404 {
+			config.ReportHttpErrorAsWarning(ctx, &resp.Diagnostics, "An error occurred while getting the SpAuthenticationPolicyContractMappingResource", err, httpResp)
+			resp.State.RemoveResource(ctx)
+		} else {
+			config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while getting the  SpAuthenticationPolicyContractMappingResource", err, httpResp)
+		}
+	}
+	// Log response JSON
+	_, responseErr := apiReadSpAuthenticationPolicyContractMappingResource.MarshalJSON()
+	if responseErr != nil {
+		diags.AddError("There was an issue retrieving the response of the SpAuthenticationPolicyContractMappingResource: %s", responseErr.Error())
+	}
+
+	// Read the response into the state
+	diags = readSpAuthenticationPolicyContractMappingResourceResponse(ctx, apiReadSpAuthenticationPolicyContractMappingResource, &state)
+	resp.Diagnostics.Append(diags...)
+	// Set refreshed state
+	diags = resp.State.Set(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+}
+
+// Update updates the resource and sets the updated Terraform state on success.
+func (r *spAuthenticationPolicyContractMappingResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+
+	var plan spAuthenticationPolicyContractMappingResourceModel
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	attributeContractFulfillment := &map[string]client.AttributeFulfillmentValue{}
```

Comment:
It looks like a lot of this is the same between this update method and the create method. Maybe some of it could be moved to a separate method to avoid the repetition.